spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java

```diff
@@ -153,67 +143,81 @@ public RestTemplate getContainerRestTemplate(boolean skipSslVerification, boolea
+	/**
+	 * Amazon, Azure and Custom Container Registry services require special treatment for the Authorization headers when the
+	 * HTTP request are forwarded to 3rd party services.
+	 *
+	 * Amazon:
+	 *   The Amazon S3 API supports two Authentication Methods (https://amzn.to/2Dg9sga):
+	 *   (1) HTTP Authorization header and (2) Query string parameters (often referred to as a pre-signed URL).
+	 *
+	 *   But only one auth mechanism is allowed at a time. If the http request contains both an Authorization header and
+	 *   an pre-signed URL parameters then an error is thrown.
+	 *
+	 *   Container Registries often use AmazonS3 as a backend object store. If HTTP Authorization header
+	 *   is used to authenticate with the Container Registry and then this registry redirect the request to a S3 storage
+	 *   using pre-signed URL authentication, the redirection will fail.
+	 *
+	 *   Solution is to implement a HTTP redirect strategy that removes the original Authorization headers when the request is
+	 *   redirected toward an Amazon signed URL.
+	 *
+	 * Azure:
+	 *   Azure have same type of issues as S3 so header needs to be dropped as well.
+	 *   (https://docs.microsoft.com/en-us/azure/container-registry/container-registry-faq#authentication-information-is-not-given-in-the-correct-format-on-direct-rest-api-calls)
+	 *
+	 * Custom:
+	 *   Custom Container Registry may have same type of issues as S3 so header needs to be dropped as well.
+	 *
+	 * @author Adam J. Weigold
+	 * @author Janne Valkealahti
+	 * @author Christian Tzolov
+	 * @author Cheng Guan Poh
+	 * @author Corneil du Plessis
+	 */
+	private HttpClient httpClientBuilder(boolean skipSslVerification) {
+
+        try {
+			SslContextBuilder builder = skipSslVerification ? SslContextBuilder.forClient().trustManager(InsecureTrustManagerFactory.INSTANCE) : SslContextBuilder.forClient();
+			SslContext sslContext = builder.build();
+			HttpClient client = HttpClient.create().secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+
+			return client.followRedirect(true, (entries, httpClientRequest) -> {
+						HttpHeaders httpHeaders = httpClientRequest.requestHeaders();
+						removeAuthorization(httpHeaders);
+						removeAuthorization(entries);
+						httpClientRequest.headers(httpHeaders);
+					});
+        } catch (SSLException e) {
+            throw new RuntimeException(e);
+        }
 	}
-	private HttpClientBuilder httpClientBuilder(SSLContext sslContext) {
-		// Register http/s connection factories
-		Lookup<ConnectionSocketFactory> connSocketFactoryLookup = RegistryBuilder.<ConnectionSocketFactory> create()
-			.register("https", new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE))
-			.register("http", new PlainConnectionSocketFactory())
-			.build();
-		return HttpClients.custom()
-			.setConnectionManager(new BasicHttpClientConnectionManager(connSocketFactoryLookup));
+
+	private static void removeAuthorization(HttpHeaders headers) {
```

Comment:
This impl is missing all of the variants described in this methods javadocs ([above](https://github.com/spring-cloud/spring-cloud-dataflow/pull/5996/files#diff-d659fdfefb699e093be40defb784f08e85441f3a595328ee597aa899184ef6b4R152-R174)) which is the meat/potatoes of the deleted request strategy `DropAuthorizationHeaderRequestRedirectStrategy`.

Reply:
Fixed creation of HttpClient